### PR TITLE
Uses Playwright for integration tests, plus other improvements.

### PR DESCRIPTION
**Pull request by nihiliad**

### CODE REVIEW

File: uri-factory/index.js
Comment (mberkowski) [inline uri-factory/index.js:16]: Any chance of logging unsupported params here? (which I guess would be the `else` case)
This doesn't have to hold up rolling these changes if it's even practical

**Review by mberkowski — APPROVED** [review]
Approve - it would be cool if we could log attempts to use bad parameters, but it also might result in a lot of junk from hacky server probes.

File: uri-factory/index.js
Comment (nihiliad) [reply uri-factory/index.js:16]: I had almost exactly the same thoughts. I want to revisit unsupported params in general (see issue #4), but I don't want that to hold up this release, which fixes some things that are broken in production. At least for now these code changes make what we're doing more explicit and self-documenting.

**Pull request merged**
